blog/2023-03-08-split-plugin-packages.md

```diff
@@ -0,0 +1,120 @@
+---
+slug: split-plugin-packages
+title: Upcoming Chnages to Plugins in Native Packages
+description: Upcoming Changes to Plugins in Native Packages
+authors: ferroin
+tags: [installation, native-packages, updates, linux, collectors, engineering]
+keywords: [installation, native-packages, updates, linux, collectors, engineering]
+
+---
+
+At Netdata, we’re committed to trying to make Netdata work as well as possible for our users. Sometimes though,
+that means changing things in ways that aren’t exactly seamless. Such a change is coming soon for users of our
+native DEB and RPM packages, and this blog post will explain what’s happening, why we’re doing it, and what
+it means for our users.
+
+<!-- truncate -->
+
+## What’s changing?
+
+Starting shortly after the v1.39.0 release of the Netdata Agent, we will be splitting most of our external
+data collection plugins out to their own individual packages instead of bundling them all in the main `netdata`
+package. We already have this type of split for our CUPS and FreeIPMI plugins, and this new change will extend
+that to also provide separate packages for the following plugins:
+
+- The `go.d` plugin (in the new `netdata-plugin-go` package)
+- The `python.d` plugin (in the new `netdata-plugin-python` package)
+- The `charts.d` plugin (in the new `netdata-plugin-chartsd` package)
+- The `slabinfo` plugin (in the new `netdata-plugin-slabinfo` package)
+- The `perf` plugin (in the new `netdata-plugin-perf` package)
+- The `nfacct` plugin (in the new `netdata-plugin-nfacct` package)
+- The eBPF plugin (in the new `netdata-plugin-ebpf` and `netdata-ebpf-legacy-code` packages)
+- The apps plugin (in the new `netdata-plugin-apps` package)
+
+## Why are we making this change?
+
+With our current approach to packaging Netdata, we actually ship a _lot_ of code that most users never actually use.
+
+For our users, this means that:
+
+- Netdata takes up more disk space than it really needs to.
```

Comment:
Any easy way to quantify the impact here on some sort of "typical" installation? Just curious if is like a major major change or more like incremental best practice type improvements so maybe not worth lots of effort to try get some representative number.

Reply:
In general, it’s not likely to be more than a few kB to a few dozen MB.